drizzlepac/haputils/catalog_utils.py

```diff
@@ -1386,19 +1394,16 @@ def identify_sources(self, **pars):
             log.info("RickerWavelet kernel X- and Y-dimension: {}".format(self._rw2d_size))
             log.info("Percentage limit on biggest source (criterion for  RickerWavelet kernel): {}".format(100.0 * self._rw2d_biggest_source))
             log.info("Percentage limit on source fraction over the image (criterion for RickerWavelet kernel): {}".format(100.0 * self._rw2d_source_fraction))
+            log.info("Percentage limit on biggest source deblending limit: {}".format(100.0 * self._bs_deblend_limit))
+            log.info("Percentage limit on source fraction deblending limit: {}".format(100.0 * self._sf_deblend_limit))
             log.info("")
             log.info("{}".format("=" * 80))
 
             # Get the SCI image data
             imgarr = copy.deepcopy(self.image.data)
 
-            # Compute the threshold to use for source detection
-            threshold = self.compute_threshold(self._nsigma, self.image.bkg_background_ra, self.image.bkg_rms_ra)
-
-            # Define smoothing kernel which overrides the use of the custom kernel
-            sigma = 1.0 * gaussian_fwhm_to_sigma
-            g2d_kernel = Gaussian2DKernel(sigma, x_size=3, y_size=3)
-            g2d_kernel.normalize()
+            # Custom or Gaussian kernel depending upon the results of CatalogImage build_kernel()
+            g2d_kernel = self.image.kernel
```

Comment:
Would it be possible for 'self.image.kernel' to not get defined?  If so, the old code may need to be used as a default case.

Reply:
The self.kernel, self.kernel_fwhm, and self.kernel_psf are attributes of the CatalogImage class, and represent the Custom (image-specific) or Gaussian kernel, as appropriate.  The self.image is an instantiation of CatalogImage for the HapCatalogs base class and is a parameter used in the constructor for the HAPSegmentCatalog/HAPPointCatalog.  Fundamentally, the kernel is derived in the build_auto_kernel() function of the astrometric_utils.py and seems to always be define deliberately.

If you still think this is risky, I can add the old code back as default.

Reply:
This just stood out as a significant change, and it wasn't immediately obvious that this was always going to be defined.  Your evaluation seems to verify that these attributes are always defined, so I don't see any problem with this part of the code.